auth/v1.js

```diff
@@ -315,11 +316,112 @@ api.declare({
     return res.status(404).json({message: "Client not found!"});
   }
 
-  // Reset accessToken
+  let added = _.without.apply(_, [input.scopes].concat(client.scopes));
+  if (!req.satisfies([added])) {
```

Comment:
btw, solution outlined in the bug you NI'ed me on was pretty awesome go with that :)